src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -537,11 +529,13 @@ private static Hashtable SplitConnectionString(string connectionString, Hashtabl
                     }
                     DebugTraceKeyValuePair(keyname, keyvalue, synonyms);
 
-                    string realkeyname = ((null != synonyms) ? (string)synonyms[keyname] : keyname);
+                    string synonym;
+                    string realkeyname = null != synonyms && synonyms.TryGetValue(keyname, out synonym) ? synonym : keyname;
```

Comment:
There is a slight behavior change from earlier. 
Earlier the behavior was, from synonyms provide the value at index `keyname` even if it doesn't exist for a non-null hashtable.

Now it is, If the value is not present then return the `keyname`. 

Have you analyzed this impact?